app/controllers/export_controller.rb

```diff
@@ -59,19 +84,57 @@ def excel_comprehensive_patients
   end
 
   def excel_full_history_patients
-    redirect_to(root_url) && return unless current_user.can_export?
+    # Verify params
+    redirect_to(root_url) && return unless %w[purgeable all].include?(params[:scope])
 
-    type = "full_history_#{params[:scope] == 'purgeable' ? 'purgeable' : 'all'}"
+    export_type = "full_history_#{params[:scope]}".to_sym
 
-    if current_user.export_receipts.where(export_type: type).where('created_at > ?', 1.hour.ago).exists?
+    if current_user.export_receipts.where(export_type: export_type).where('created_at > ?', 1.hour.ago).exists?
       render json: { message: 'You have already initiated an export of this type in the last hour. Please try again later.' }.to_json, status: 401
     else
       # Clear out old receipts and create a new one
-      current_user.export_receipts.where(export_type: type).destroy_all
-      ExportReceipt.create(user_id: current_user.id, export_type: type)
+      current_user.export_receipts.where(export_type: export_type).destroy_all
+      ExportReceipt.create(user_id: current_user.id, export_type: export_type)
 
       # Spawn job to handle export
-      ExportJob.perform_later(current_user.id, type)
+      config = {
+        user_id: current_user.id,
+        export_type: export_type,
+        format: 'xlsx',
+        filename: "Sara-Alert-#{params[:scope] == 'purgeable' ? 'Purge-Eligible' : 'Full'}-Export",
+        filename_data_type: true,
+        separate_files: true,
+        data: {
+          patients: {
+            checked: FULL_HISTORY_PATIENTS_FIELDS,
+            headers: FULL_HISTORY_PATIENTS_HEADERS,
+            query: params[:scope] == 'purgeable' ? { purgeable: true } : {},
+            name: 'Monitorees',
+            tab: 'Monitorees List'
+          },
+          # assessment fields and headers need to be duplicated because they may be modified
```

Comment:
good question, it's because this only applies to full history exports where the assessment headers and fields don't include the headers and fields for the symptoms (which you could probably already tell). For patients though, the fields and headers will not be modified in this type of export because the race fields are already expanded:

https://github.com/SaraAlert/SaraAlert/blob/SARAALERT-950/app/lib/import_export.rb#L56